### PR TITLE
Add experimental Music Skill

### LINKED ISSUE #2005: Ask to play artist and send to Spotify
Create VA skill capable of capturing an Artist Name and passing it down Spotify music service launching the Spotify app in a deep link.

### PR DESCRIPTION
**Pull request by ryanisgrig**
### Purpose
*What is the context of this pull request? Why is it being done?*
Close #2005 enabling users to ask to play an artists and play via Spotify

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
* Using SpotifyAPI-NET package to facilitate Spotify client use

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
No, experimental skill

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [x] I have updated related documentation - will do separately on master branch

#### Bots
- [x] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [x] I have replicated language model changes across the English and Chinese `.lu` files and validated that deployment is successful

### CODE REVIEW

**Review by lzc850612 — APPROVED** [review]

File: skills/src/csharp/experimental/musicskill/Deployment/Resources/LU/en/MusicSkill.lu [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+- what is fox news' top story
+- what is good friday
+- what is this election results
+- www.read.amazon.com
+- www.twitter.com
+- www.wikipedia.com please
+
+
+## PlayMusic
+- play ariana grande
```
Comment (darrenj) [inline skills/src/csharp/experimental/musicskill/Deployment/Resources/LU/en/MusicSkill.lu:143]: we should expand these utterances in the next update

File: skills/src/csharp/experimental/musicskill/Dialogs/PlayMusicDialog.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        private async Task<DialogTurnResult> GetAndSendMusicResult(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            var state = await StateAccessor.GetAsync(stepContext.Context);
+            var intent = state.LuisResult.TopIntent().intent;
+            var entities = state.LuisResult.Entities;
+
+            // Extract query entity to search against Spotify for
+            var searchQuery = entities.Artist_Any[0];
```
Comment (darrenj) [inline skills/src/csharp/experimental/musicskill/Dialogs/PlayMusicDialog.cs:52]: what if no entity is found? Should make it defensive and perhaps prompt if not?

File: skills/src/csharp/experimental/musicskill/Dialogs/PlayMusicDialog.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+            // Get music api client
+            var client = await GetSpotifyWebAPIClient(Settings);
+
+            // Search library
+            var searchItems = await client.SearchItemsEscapedAsync(searchQuery, SearchType.All, 5);
+
```
Comment (darrenj) [inline skills/src/csharp/experimental/musicskill/Dialogs/PlayMusicDialog.cs:59]: exception handling for service connectivity issues?

File: skills/src/csharp/experimental/musicskill/readme.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Please refer to the [Skill Template documentation](http://aka.ms/virtualassistantdocs) for deployment and customization instructions.
```
Comment (darrenj) [inline skills/src/csharp/experimental/musicskill/readme.md:1]: Perhaps we need to add where/how you get a Spotify key settings in app.config?

**Review by darrenj — CHANGES_REQUESTED** [review]

File: skills/src/csharp/experimental/musicskill/readme.md (hunk 4 above)
Comment (ryanisgrig) [reply skills/src/csharp/experimental/musicskill/readme.md:1]: I have it in the experimental docs page for now

**Pull request merged**